pkg/storegateway/indexcache/cache.go

```diff
@@ -76,6 +86,33 @@ type IndexCache interface {
 	FetchLabelValues(ctx context.Context, userID string, blockID ulid.ULID, labelName string, matchersKey LabelMatchersKey) ([]byte, bool)
 }
 
+// PostingsKey represents a canonical key for a []storage.SeriesRef slice
+type PostingsKey string
+
+// CanonicalPostingsKey creates a canonical version of PostingsKey
+func CanonicalPostingsKey(postings []storage.SeriesRef) PostingsKey {
+	sorted := make([]byte, len(postings)*8)
```

Comment:
As discussed on Slack, I have 3 feedbacks:
1. We can assume postings are always sorted (we can make it a requirement)
2. Sorting octects is != than sorting integers. Sorting octects will easily lead to collisions.
3. Can we avoid this `make([]byte)` at all, and use unsafe to get a reference to the underlying memory area (as a `[]byte`) that we then directly pass to the hasher `Write()`?

Reply:
we agreed to do this in a follow-up PR